hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/NodeEndpoint.java

```diff
@@ -139,18 +139,18 @@ public Response getDatanodes() {
       }
 
       DatanodeInfo dnInfo = (DatanodeInfo) datanode;
-      datanodes.add(builder.withHostname(hostname)
+      datanodes.add(builder.withHostname(nodeManager.getHostName(datanode))
           .withDatanodeStorageReport(storageReport)
           .withLastHeartbeat(nodeManager.getLastHeartbeat(datanode))
           .withState(nodeState)
           .withOperationalState(nodeOpState)
           .withPipelines(pipelines)
           .withLeaderCount(leaderCount.get())
           .withUUid(datanode.getUuidString())
-          .withVersion(datanode.getVersion())
-          .withSetupTime(datanode.getSetupTime())
-          .withRevision(datanode.getRevision())
-          .withBuildDate(datanode.getBuildDate())
+          .withVersion(nodeManager.getVersion(datanode))
```

Comment:
I see that the in memory DN details are not seeded with the DB DN details on startup. This means that dead datanode information will never show up in Recon UI. While SCM is supposed to make use of only 'Live' nodes, Recon has to keep track of **all** nodes in the system.

Reply:
I saw the original `loadingExistingDB` is calling the method of `register` for all existing datanodes, so it will be added to in-mem DB on startup.